--- src.orig/hpl/testing/ptest/HPL_pdtest.c.orig	2008-02-13 12:46:45.867835000 -0700
+++ src/hpl/testing/ptest/HPL_pdtest.c	2008-02-13 12:47:31.992583000 -0700
@@ -165,7 +165,7 @@ void HPL_pdtest
 /*
  * Allocate dynamic memory
  */
-   vptr = (void*)HPCC_malloc( (ALGO->align + (mat.ld+1)*(mat.nq))*sizeof(double) );
+   vptr = (void*)HPCC_malloc((long)(ALGO->align + (long)(mat.ld+1)*(mat.nq))*sizeof(double) );
    info[0] = (vptr == NULL); info[1] = myrow; info[2] = mycol;
    (void) HPL_all_reduce( (void *)(info), 3, HPL_INT, HPL_max,
                           GRID->all_comm );
