--- src/miranda_io.f90.orig     2007-02-06 15:03:05.000000000 -0700
+++ src/miranda_io.f90      2009-05-20 12:39:25.000000000 -0600
@@ -69,12 +69,13 @@

 !     check environment for file name, then broadcast from root task
       if (mynod == 0) then
-        call getenv('MIRANDA_IO_FNAME', value=home)
+!        call getenv('MIRANDA_IO_FNAME', value=home)
+        call getenv('MIRANDA_IO_FNAME', home)
         if (trim(home) /= '') then
           basefname = trim(home)
         endif
       endif
-      call MPI_BCAST(basefname, 1024, MPI_CHAR, 0, MPI_COMM_WORLD, ierr);
+      call MPI_BCAST(basefname, 1024, MPI_CHARACTER, 0, MPI_COMM_WORLD, ierr);

       if( mynod == 0 ) then
         print *, 'Fortran I/O test emulating Bill Cabots code and bz4410'
